aspnetcore/mvc/razor-pages/razor-pages-convention-features/sample/Factories/AddHeaderWithFactory.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace ModelProvidersSample.Factories
+{
+    #region snippet1
+    public class AddHeaderWithFactory : IFilterFactory
```

Comment:
You might not need a factory. You could just add the filter directly. FilterFactories are useful if you want to do some DI \ state management shenanigans.

Reply:
Adding a filter directly is also in the topic. I tried to show *everything* that they could do, including this approach.

Reply:
Let me see if I can surface these by cross-linking the two of them.

Reply:
Ah ... I see why I didn't do that earlier. They're stacked in the topic ... **Configure a filter factory** comes immediately after **Configure a filter**, so they're so close to each other that there's probably no point to cross-linking.

Reply:
Is there value in having `Configure a filter factory` topic? Seems like if you're familiar with filters, the Configure a filter topic should give you enough details to figure out the latter.